octodns_edgecenter/__init__.py

```diff
@@ -167,11 +182,21 @@ def _build_pools(self, record, default_pool_name, value_transform_fn):
             value = {"value": value_transform_fn(rr["content"][0])}
             countries = meta.get("countries", []) or []
             continents = meta.get("continents", []) or []
+            # weight = meta.get("weight", []) or []
 
             if meta.get("default", False):
                 pools[default_pool_name]["values"].append(value)
                 defaults.append(value["value"])
                 continue
+            elif meta.get("weight", 0) > 0:
+                # elif len(weight) > 0:
+                value_weight = {
+                    "value": value_transform_fn(rr["content"][0]),
+                    "weight": meta.get("weight", 0),
```

Comment:
Fwiw you shouldn't need a default here since you're in an `elif` that checked for a non-default value `> 0`,

```suggestion
                    "weight": meta["weight"],
```